odoo/addons/base/models/ir_qweb_fields.py

```diff
@@ -609,17 +609,25 @@ def get_available_options(self):
             width=dict(type='integer', string=_('Width'), default_value=600),
             height=dict(type='integer', string=_('Height'), default_value=100),
             humanreadable=dict(type='integer', string=_('Human Readable'), default_value=0),
+            quiet=dict(type='integer', string='Quiet', default_value=1),
+            mask=dict(type='string', string='Mask', default_value='')
         )
         return options
 
     @api.model
     def value_to_html(self, value, options=None):
+        if not value:
+            return ''
         barcode_symbology = options.get('symbology', 'Code128')
         barcode = self.env['ir.actions.report'].barcode(
             barcode_symbology,
             value,
-            **{key: value for key, value in options.items() if key in ['width', 'height', 'humanreadable']})
-        return u'<img src="data:png;base64,%s">' % base64.b64encode(barcode).decode('ascii')
+            **{key: value for key, value in options.items() if key in ['width', 'height', 'humanreadable', 'quiet', 'mask']})
+        img_attributes = {k[4:]: v for k, v in options.items() if k.startswith('img_')}
+        if 'alt' not in img_attributes:
+            img_attributes.update({'alt': 'Barcode %s' % value})
+        attributes = ' '.join(['%s="%s"' % attrs for attrs in img_attributes.items()])
```

Comment:
that may or may not be exploitable, but it's a cardinal sin either way - please see https://github.com/odoo/enterprise/pull/15297#issuecomment-768985663 and fix it in master too ;-)

Reply:
And at that point and with that sort of complexity it'd probably make more sense to just create an lxml Element and serialise it back with html.tostring or something no?